examples/timer_gpout/timer_gpout_main.c

```diff
@@ -0,0 +1,306 @@
+  int fd_timer;
+  int fd_gpout;
+  int ret;
+  bool state = false;
+
+  /* Indicate that the deamon is running */
+
+  timer_gpout_daemon_started_g = true;
+  printf("timer_gpout_daemon: timer_gpout_daemon started\n");
+
+  /* Open the timer device */
+
+  printf("Open %s\n", devtim_g);
+
+  fd_timer = open(devtim_g, O_RDONLY);
+  if (fd_timer < 0)
+    {
+      int errcode = errno;
+      printf("timer_gpout_daemon: Failed to open %s: %d\n",
+              devtim_g, errcode);
+      return EXIT_FAILURE;
+    }
+
+  /* Open the GPIO driver */
+
+  printf("Open %s\n", devgpout_g);
+
+  fd_gpout = open(devgpout_g, O_RDWR);
+  if (fd_gpout < 0)
+    {
+      int errcode = errno;
+      printf("timer_gpout_daemon: Failed to open %s: %d\n",
+              devgpout_g, errcode);
+      close(fd_timer);
+      return EXIT_FAILURE;
+    }
+
+  /* Show the timer status before setting the timer interval */
+
+  timer_status(fd_timer);
+
+  /* Set the timer interval */
+
+  printf("Set timer interval to %lu\n",
+         (unsigned long)CONFIG_EXAMPLES_TIMER_GPOUT_INTERVAL);
+
+  ret = ioctl(fd_timer, TCIOC_SETTIMEOUT,
+              CONFIG_EXAMPLES_TIMER_GPOUT_INTERVAL);
+  if (ret < 0)
+    {
+      int errcode = errno;
+      printf("timer_gpout_daemon: Failed to set the timer interval: %d\n",
+             errcode);
+      goto errout;
+    }
+
+  /* Show the timer status before attaching the timer handler */
+
+  timer_status(fd_timer);
+
+  /* Configure the signal set for this task */
+
+  sigemptyset(&set);
+  sigaddset(&set, CONFIG_EXAMPLES_TIMER_GPOUT_SIGNO);
+
+  /* Configure the timer notifier to receive a signal when timeout occurs.
+   * Inform the PID of the process that will be notified by the internal
+   * timer callback, and configure the event to describe the way this
+   * task will be notified. In this case, through a signal.
+   */
+
+  printf("Configure the timer notification.\n");
+
+  notify.pid   = getpid();
+  notify.event.sigev_notify = SIGEV_SIGNAL;
+  notify.event.sigev_signo  = CONFIG_EXAMPLES_TIMER_GPOUT_SIGNO;
+  notify.event.sigev_value.sival_ptr = NULL;
+
+  ret = ioctl(fd_timer, TCIOC_NOTIFICATION,
+              (unsigned long)((uintptr_t)&notify));
+  if (ret < 0)
+    {
+      int errcode = errno;
+      printf("timer_gpout_daemon: Failed to set the timer handler: %d\n",
+             errcode);
+      goto errout;
+    }
+
+  /* Start the timer */
+
+  printf("Start the timer\n");
+
+  ret = ioctl(fd_timer, TCIOC_START, 0);
+  if (ret < 0)
+    {
+      int errcode = errno;
+      printf("timer_gpout_daemon: Failed to start the timer: %d\n", errcode);
+      goto errout;
+    }
+
+  while (1)
+    {
+      ret = sigwaitinfo(&set, &value);
+      if (ret < 0)
+        {
+          int errcode = errno;
+          printf("timer_gpout_daemon: ERROR: sigwaitinfo() failed: %d\n",
+                 errcode);
+          goto errout;
+        }
+
+      /* Change the gpout state */
+
+      state = ~state;
```

Comment:
```suggestion
      state = !state;
```
The tilde is a bit operation of a boolean value.  It will toggle from 0 (false) and 0xffffffff (invalid boolean value).  The exclamation point is the correct operator for a boolean and will toggle from 0 (false) to 1 (true).

Reply:
Ok.
Fixed.